src/Extractor/PDFParserTextExtractor.php

```diff
@@ -46,15 +53,33 @@ public function supportsMime($mime): bool
         );
     }
 
+    /**
+     * Get instance of PDF parser
+     */
+    protected function getParser(): Parser
+    {
+        $pdfParser = new Parser();
+
+        $this->extend('onInitParser', $pdfParser);
+
+        return $pdfParser;
+    }
+
     /**
      * @inheritDoc
      */
     public function getContent($file): string
     {
-        $pdfParser = new Parser();
+        $pdfParser = $this->getParser();
+
         try {
             $path = $file instanceof File ? self::getPathFromFile($file) : $file;
-            $text = $pdfParser->parseFile($path)->getText();
+            $text = $pdfParser->parseFile($path)->getText(self::config()->get('pages_to_parse'));
+
+            // Remove new lines and spaces
+            $text = str_replace("\n", ' ', $text);
+            $text = preg_replace('/\s+/', ' ', $text);
```

Comment:
In the interest of allowing people to have this behave in the same way it was behaving, could we add a config flag for this too please? Happy for it to default to "remove excess whitespace" but I'd like to have the option, in case people are doing some weird post-parsing newline-splitting or something like that.

Reply:
Updated :)